packages/fetch-and-compile/test/fetch.test.ts

```diff
@@ -0,0 +1,81 @@
+import assert from "assert";
+import { describe, it } from "mocha";
+import Config from "@truffle/config";
+import { fetchAndCompile } from "../lib/index";
+import axios from "axios";
+import sinon from "sinon";
+const fixture: any = require("./fixture.js");
+
+function axiosMock(url: string, address: string, data: object) {
+  sinon.stub(axios, 'get').withArgs(url, {
+    params: {
+      module: "contract",
+      action: "getsourcecode",
+      address: address,
+      apikey: ''
+    },
+    responseType: "json",
+    maxRedirects: 50
+  }).returns(Promise.resolve({ data: data }))
+}
+afterEach(()=>
+    //@ts-ignore
+    //restoring stub
+    axios.get.restore()
+)
+describe("fetchAndCompile", () => {
+  it('verifes contract from mainnet', async () => {
+    const config = Config.default().merge({
+      networks: {
+        mainnet: {
+          network_id: 1
+        }
+      },
+      network: "mainnet",
+    });
+    //asserting that mainnet url and contract address is passed as args
+    axiosMock("https://api.etherscan.io/api", '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D', fixture.mainnetData)
+    //@ts-ignore
+    axios.get.callThrough();
+    const result = await fetchAndCompile('0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D', config)
```

Comment:
The contract address can be a variable.
```
const address = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D";
axiosMock(..., address, ...)
fetchAndCompile(address, ...)
```

```

Reply:
Yup yup a variable makes a ton of more sense. Made this change! Thanks.